app/components/channel_drawer/channel_drawer.js

```diff
@@ -128,20 +146,136 @@ export default class ChannelDrawer extends PureComponent {
         });
     };
 
-    render() {
+    swiperPageSelectedAndroid = (event) => {
+        this.swiperIndex = event.nativeEvent.position;
+    };
+
+    swiperPageSelectedIos = (e, state, context) => {
+        this.swiperIndex = context.state.index;
+    };
+
+    showTeams = () => {
+        const teamsCount = Object.keys(this.props.myTeamMembers).length;
+        if (this.swiperIndex === 1 && teamsCount > 1) {
+            if (Platform.OS === 'android') {
+                this.swiper.setPage(0);
+            } else {
+                this.swiper.scrollBy(-1, true);
+            }
+        }
+    };
+
+    resetDrawer = () => {
+        const teamsCount = Object.keys(this.props.myTeamMembers).length;
+        if (this.swiperIndex === 0 && teamsCount > 1) {
+            if (Platform.OS === 'android') {
+                this.swiper.setPageWithoutAnimation(1);
+            } else {
+                this.swiper.scrollBy(1, false);
+            }
+        }
+    };
+
+    renderAndroid = (theme, teams, channelsList, showTeams) => {
```

Comment:
Would it be possible to make a component that wraps both ViewPagerAndroid and Swiper so that we can better isolate the platform-specific code for these?